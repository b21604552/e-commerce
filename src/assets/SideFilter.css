.side-filter-wrapper {
    width: 206px;
    margin-left: 80px;
    margin-right: 16px;
    margin-top: 16px;
    padding-top: 16px;
}

.side-filter-element {
    margin-bottom: 24px;
}

.filter-element-header {
    font-style: normal;
    font-weight: 500;
    font-size: 16px;
    line-height: 24px;
    color: #484848;
    margin-bottom: 12px;
}

.filter-option-wrapper {

}

.filter-option {
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 24px;
    color: #646464;
    cursor: pointer;
}

.filter-option:hover {
    color: #FF6000;
}

.filter-option-active {
    color: #FF6000;
}