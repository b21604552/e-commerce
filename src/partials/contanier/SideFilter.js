import React, { useEffect, useState } from 'react'
import { setOrderValue } from '../../services/redux/actions/GeneralAction'
import { reduxStore } from '../../services/redux/ReduxIndex'

export default function SideFilter (props) {
  const { orderValue } = reduxStore.getState().GeneralReducer
  const [filterOptions, setFilterOptions] = useState(props.filterOptions)
  const [selectedColor, setSelectedColor] = useState(null)
  const [selectedBrand, setSelectedBrand] = useState(null)

  useEffect(() => {
    setFilterOptions(props.filterOptions)
  }, [props.filterOptions])

  function toggleSelectedColor (color, i) {
    if (i === selectedColor) {
      props.setFilterColor(null)
      setSelectedColor(null)
    } else {
      props.setFilterColor(color)
      setSelectedColor(i)
    }
  }

  function toggleSelectedBrand (brand, i) {
    if (i === selectedBrand) {
      props.setFilterBrand(null)
      setSelectedBrand(null)
    } else {
      props.setFilterBrand(brand)
      setSelectedBrand(i)
    }
  }

  function setOrder (field, type, i) {
    if (i === orderValue) {
      props.setOrder(null)
      setOrderValue('')
    } else {
      props.setOrder(field, type)
      setOrderValue(i)
    }
  }

  return (
    <div className={'side-filter-wrapper'}>
      <div className={'side-filter-element'}>
        <div className={'filter-element-header'}>
          Renk
        </div>
        <div className={'filter-option-wrapper'}>
          {
            filterOptions.color.map((color, i) => (
              <div key={'color-' + i} className={i === selectedColor ? ('filter-option filter-option-active') : ('filter-option')}
                   onClick={() => toggleSelectedColor(color.text, i)}>{color.text} ({color.value})</div>
            ))
          }
        </div>
      </div>
      <div className={'side-filter-element'}>
        <div className={'filter-element-header'}>
          Sıralama
        </div>
        <div className={'filter-option-wrapper'}>
          <div id={'order-0'} className={orderValue === '0' ? ('filter-option filter-option-active') : ('filter-option')} onClick={() => setOrder('discountedPrice', 'asc', '0')}>
            En Düşük Fiyat
          </div>
          <div id={'order-1'} className={orderValue === '1' ? ('filter-option filter-option-active') : ('filter-option')} onClick={() => setOrder('discountedPrice', 'desc', '1')}>
            En Yüksek Fiyat
          </div>
          <div id={'order-2'} className={orderValue === '2' ? ('filter-option filter-option-active') : ('filter-option')} onClick={() => setOrder('title', 'asc', '2')}>
            En Yeniler (A>Z)
          </div>
          <div id={'order-3'} className={orderValue === '3' ? ('filter-option filter-option-active') : ('filter-option')} onClick={() => setOrder('title', 'desc', '3')}>
            En Yeniler (Z>A)
          </div>
        </div>
      </div>
      <div className={'side-filter-element'}>
        <div className={'filter-element-header'}>
          Marka
        </div>
        <div className={'filter-option-wrapper'}>
          {
            filterOptions.brand.map((brand, i) => (
              <div key={'brand-' + i} className={i === selectedBrand ? ('filter-option filter-option-active') : ('filter-option')}
                   onClick={() => toggleSelectedBrand(brand.text, i)}>{brand.text} ({brand.value})</div>
            ))
          }
        </div>
      </div>
    </div>
  )
}