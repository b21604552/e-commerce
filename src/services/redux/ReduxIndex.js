import { createStore } from 'redux'
import reducerCombined from './reducerCombined'

export const reduxStore = createStore(reducerCombined)
