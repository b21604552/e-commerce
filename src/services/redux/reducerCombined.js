import { combineReducers } from 'redux'
import GeneralReducer from './reducers/GeneralReducer'

const reducerCombined = combineReducers({
  GeneralReducer: GeneralReducer
})

export default reducerCombined
