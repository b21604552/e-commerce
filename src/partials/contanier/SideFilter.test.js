import React from 'react'
import ReactDOM from 'react-dom'
import SideFilter from './SideFilter'
import renderer from 'react-test-renderer'
import { reduxStore } from '../../services/redux/ReduxIndex'
import { Provider } from 'react-redux'

const filterOptions = {
  'color': [
    {
      'text': 'Siyah',
      'value': 10
    },
    {
      'text': 'Lacivert',
      'value': 10
    },
    {
      'text': 'SarÄ±',
      'value': 10
    },
    {
      'text': 'Beyaz',
      'value': 10
    }
  ],
  'brand': [
    {
      'text': 'Apple',
      'value': 16
    },
    {
      'text': 'Samsung',
      'value': 12
    },
    {
      'text': 'Xiaomi',
      'value': 4
    },
    {
      'text': 'Huawei',
      'value': 4
    },
    {
      'text': 'LG',
      'value': 4
    }
  ]
}

describe('SideFilter Test', () => {
  let page = (
    <Provider store={reduxStore}>
      <SideFilter filterOptions={filterOptions}/>
    </Provider>
  )

  it('Renders SideFilter without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(page, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('Renders SideFilter correctly', () => {
    const div = renderer.create(page).toJSON()
    expect(div).toMatchSnapshot()
  })
})