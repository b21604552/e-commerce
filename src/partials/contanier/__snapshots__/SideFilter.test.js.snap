// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideFilter Test Renders SideFilter correctly 1`] = `
<div
  className="side-filter-wrapper"
>
  <div
    className="side-filter-element"
  >
    <div
      className="filter-element-header"
    >
      Renk
    </div>
    <div
      className="filter-option-wrapper"
    >
      <div
        className="filter-option"
        onClick={[Function]}
      >
        Siyah
         (
        10
        )
      </div>
      <div
        className="filter-option"
        onClick={[Function]}
      >
        Lacivert
         (
        10
        )
      </div>
      <div
        className="filter-option"
        onClick={[Function]}
      >
        Sarı
         (
        10
        )
      </div>
      <div
        className="filter-option"
        onClick={[Function]}
      >
        Beyaz
         (
        10
        )
      </div>
    </div>
  </div>
  <div
    className="side-filter-element"
  >
    <div
      className="filter-element-header"
    >
      Sıralama
    </div>
    <div
      className="filter-option-wrapper"
    >
      <div
        className="filter-option"
        id="order-0"
        onClick={[Function]}
      >
        En Düşük Fiyat
      </div>
      <div
        className="filter-option"
        id="order-1"
        onClick={[Function]}
      >
        En Yüksek Fiyat
      </div>
      <div
        className="filter-option"
        id="order-2"
        onClick={[Function]}
      >
        En Yeniler (A&gt;Z)
      </div>
      <div
        className="filter-option"
        id="order-3"
        onClick={[Function]}
      >
        En Yeniler (Z&gt;A)
      </div>
    </div>
  </div>
  <div
    className="side-filter-element"
  >
    <div
      className="filter-element-header"
    >
      Marka
    </div>
    <div
      className="filter-option-wrapper"
    >
      <div
        className="filter-option"
        onClick={[Function]}
      >
        Apple
         (
        16
        )
      </div>
      <div
        className="filter-option"
        onClick={[Function]}
      >
        Samsung
         (
        12
        )
      </div>
      <div
        className="filter-option"
        onClick={[Function]}
      >
        Xiaomi
         (
        4
        )
      </div>
      <div
        className="filter-option"
        onClick={[Function]}
      >
        Huawei
         (
        4
        )
      </div>
      <div
        className="filter-option"
        onClick={[Function]}
      >
        LG
         (
        4
        )
      </div>
    </div>
  </div>
</div>
`;
